build/tasks/test.js

```diff
@@ -7,18 +7,20 @@ var sourcemaps = require('gulp-sourcemaps');
 var reportChange = require('../reportChange').reportChange;
 
 var tsProject = ts.createProject({
+  typescript: require('typescript'),
   declarationFiles: false,
   noExternalResolve: false,
-  target: 'ES5',
-  module: 'AMD'
+  target: "es5",
+  module: "amd",
+  emitDecoratorMetadata: true
 });
 
 // compile unit test TS file
 gulp.task('build-test', function () {
-  var tsResult = gulp.src([paths.unitSpecsSrc, paths.dtsSource])
+  var tsResult = gulp.src([paths.unitSpecsSrc, paths.dtsSource], {base: "src/"})
```

Comment:
Needed to add this option to make TS compiler happy. Otherwise it was giving the following error:
```shell
6059 File '../Projects/aurelia-skeleton-navigation-typescript/src/app/app.ts' is not under 'rootDir' '../Projects/aurelia-skeleton-navigation-typescript/src/test'. 'rootDir' is expected to contain all source files.
```